integration/test/basic/basic_test.go

```diff
@@ -63,25 +59,57 @@ func TestChartConfigChartsInstalled(t *testing.T) {
 		return
 	}
 
-	guestNamespace := "giantswarm"
 	guestG8sClient := g.G8sClient()
 
 	// Wait for chart configs as they may not have been created yet.
 	err := waitForChartConfigs(guestG8sClient)
 	if err != nil {
 		t.Fatalf("could not get chartconfigs %v", err)
 	}
+}
+
+func TestChartConfigPatchDeploy(t *testing.T) {
+	// These versions have no chartconfigs so we return early.
+	clusterOperatorVersion := os.Getenv("CLOP_VERSION_BUNDLE_VERSION")
+	if clusterOperatorVersion == "0.1.0" || clusterOperatorVersion == "0.2.0" {
+		return
+	}
+
+	guestG8sClient := g.G8sClient()
 
 	chartConfigList, err := guestG8sClient.CoreV1alpha1().ChartConfigs(guestNamespace).List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("could not get chartconfigs %v", err)
 	}
-
-	// At least 1 chartconfig is expected. Actual number depends on how many
-	// components have been migrated.
+	// At least 1 chartconfig is required.
 	if len(chartConfigList.Items) == 0 {
 		t.Fatalf("expected at least 1 chartconfigs: %d found", len(chartConfigList.Items))
 	}
+	chartConfigName := chartConfigList.Items[0].Spec.Chart.Name
+
+	patch := ChartConfigDeployPatch{
```

Comment:
All this is a bit wild. Maybe check https://github.com/giantswarm/e2etests on how they are structured and written there in block form with opening and closing logs. I want to prevent we write "some" spaghetti code which just goes out of hand over time and is really hard to scan.